ignite/src/test/java/org/hibernate/ogm/datastore/ignite/utils/IgniteTestConfigurationBuilder.java

```diff
@@ -0,0 +1,416 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.datastore.ignite.utils;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.ignite.IgniteCheckedException;
+import org.apache.ignite.IgniteSystemProperties;
+import org.apache.ignite.binary.BinaryObject;
+import org.apache.ignite.cache.CacheAtomicityMode;
+import org.apache.ignite.cache.CacheMode;
+import org.apache.ignite.cache.QueryEntity;
+import org.apache.ignite.cache.QueryIndex;
+import org.apache.ignite.cache.QueryIndexType;
+import org.apache.ignite.cache.affinity.rendezvous.RendezvousAffinityFunction;
+import org.apache.ignite.configuration.BinaryConfiguration;
+import org.apache.ignite.configuration.CacheConfiguration;
+import org.apache.ignite.configuration.IgniteConfiguration;
+import org.apache.ignite.internal.IgnitionEx;
+import org.apache.ignite.internal.binary.BinaryMarshaller;
+import org.apache.ignite.logger.slf4j.Slf4jLogger;
+import org.hibernate.ogm.datastore.ignite.IgniteConfigurationBuilder;
+
+/**
+ * Ignite cache configuration for tests
+ *
+ * @author Victor Kadachigov
+ */
+public class IgniteTestConfigurationBuilder implements IgniteConfigurationBuilder {
+
+	@Override
+	public IgniteConfiguration build() {
+		//disable check for new versions
+		System.setProperty( IgniteSystemProperties.IGNITE_UPDATE_NOTIFIER, Boolean.FALSE.toString() );
+		
+		IgniteConfiguration config = null;
+
+		try {
+			//config = loadFromResource( "ignite-config.xml" );
+			config = createConfig();
+		}
+		catch (Exception ex) {
+			throw new RuntimeException( ex );
+		}
+
+		return config;
+	}
+
+	private IgniteConfiguration loadFromResource(String resourceName) throws IgniteCheckedException {
+		URL url = getClass().getClassLoader().getResource( resourceName );
+		return IgnitionEx.loadConfiguration( url ).get1();
+	}
+
+	private IgniteConfiguration createConfig() {
+		IgniteConfiguration config = new IgniteConfiguration();
+
+		config.setGridName( "OgmTestGrid" );
+		config.setLocalHost( "127.0.0.1" );
```

Comment:
I don't think it's a good idea. 
I made configuration specially to run tests locally.
In case with remote server we need two configurations client and server, not a property file. 
IMHO it's too complicated.